File: recursion.hs
-- recursion.hs
import System.IO

factorial :: Int -> Int
factorial n
    | n <= 1 = 1
    | otherwise = n * factorial (n - 1)

main :: IO ()
main = do
    putStr "Enter an integer: "
    hFlush stdout
    input <- getLine
    let param = read input :: Int
    if param == 0 && input /= "0"
    then putStrLn "Invalid input"
    else print (factorial param)

File: recursion.m
% recursion.m
function result = factorial(n)
    if n <= 1
        result = 1;
    else
        result = n * factorial(n - 1);
    end
end

input = input('Enter an integer: ', 's');
param = str2num(input);
if isempty(param)
    disp('Invalid input')
else
    disp(['Factorial: ', num2str(factorial(param))])
end
